### ISSUE TITLE
Exit code for --max-downloads

### ISSUE THREAD

**Issue by weedy** [rg3/youtube-dl#10388]
As a unix user I fundamentally disagree with how #852 went but that's history now.

That said I can't imagine why you would want to exit non-zero when you use --max-downloads.

I wanted maximum 8 files, you gave me 8 files. Then exited with an error. 
How does this make sense?

**Comment by yan12125** [comment]
As a reference, this "feature" is added in 94fd3201b237b2a1d04e06dd7fb5ea4130bddded. I guess it's because script writers need to know the reason of exiting. Ping @phihag.

By the way, ```--max-downloads``` violation has already a non-default exit code 101.

**Comment by tommcintyre** [comment]
Agreed with @weedy, a nonzero exit code is not appropriate in this situation as I mentioned in another issue here: https://github.com/rg3/youtube-dl/issues/13598#issuecomment-317215157

The canonical way to do this would be via some stats/summary info, either printed out to the console or to a file in a computer-readable format.
